### ISSUE TITLE
Philipps CD-I [cdimono1] : no DIGITAL VIDEO cardtridge 

### ISSUE THREAD

**Issue by legluondunet** [mamedev/mame#1170]
Hello,

some CD-I like "7th guest" needs a DIGITAL VIDEO Cardtridge . 
You can not play this game without this extension card.
It is planned?

LGDN
Mame 0.176

**Comment by rb6502** [comment]
Yes, support for the Digital Video cartridge is planned, but first our CD-i developer is working on better understanding the system so that we can improve compatibility with regular titles.

**Comment by MooglyGuy** [comment]
Plus we have literally zero documentation on how the DVC actually worked. I've heard rumors that some of them have a DSP56k to handle the MPEG audio decoding, but that still leaves a big black box where the MPEG video decoding is. It would also be nice to know how the presence of the DVC is detected, how the CD-i console itself sends data to the DVC to be decoded, and how the decoded results are then mixed with the output from the VDSC and such. Because all of that is a total unknown right now.

While we're on the subject of the CD-i, it would be great if someone could reach out to Philips and ask them to see if they can dig up any documentation on the LEMM, CDIC, and any other CD-i related ASICs and DSPs that we currently black-box. Because as far as I know, they don't make CD-i players anymore, and that business is entirely dead, so they wouldn't be losing anything by sharing the documentation.

**Comment by stilett0** [comment]
> While we're on the subject of the CD-i, it would be great if someone could reach out to Philips and ask them to see if they can dig up any documentation on the LEMM, CDIC, and any other CD-i related ASICs and DSPs that we currently black-box.

... I thought you were going to try that.

**Comment by guy2010** [comment]
Any updates?

**Comment by MooglyGuy** [comment]
No

**Comment by Meerkov** [comment]
[cdi_loader.zip](https://github.com/user-attachments/files/17943914/cdi_loader.zip)
Here's the loader file. Just eyeballing it, it looks very short before it decides which file to load.
![image](https://github.com/user-attachments/assets/557f718e-182f-48e3-b406-aa03dc465d7a)

**Comment by MooglyGuy** [comment]
That doesn't really help much. If you can determine where it is specifically loaded into RAM, and throw MAME's disassembler at it, then that might be a reasonable starting point.

**Comment by petrij98** [comment]
I've been trying to do a bit of extra research online, but I'm having difficulty figuring out exactly what the purpose of the DVC was even after so many years. Was it an expansion pack like the N64? Was it an add-on like the Sega CD? I'd love to try and mess around with this a bit, but the answer to those questions would definitely determine how I best approach it. I'm surprised Philips hasn't tried to help make this happen themselves. The cdi.xml makes it look like a pretty decent chunk of the library needs the DVC and since they really aren't in the console space anymore, it'd be a HUGE way to get people into it long past its EOL.

**Comment by petrij98** [comment]
> [cdi_loader.zip](https://github.com/user-attachments/files/17943914/cdi_loader.zip) Here's the loader file. Just eyeballing it, it looks very short before it decides which file to load. ![image](https://github.com/user-attachments/assets/557f718e-182f-48e3-b406-aa03dc465d7a)

Just based off of what I can read, you're definitely right; it's not a ton of information between loader and t7g. A couple of things have me intrigued, though; what does nothing point to and how does 7th guest's loader file compare side by side with a non-DVC game's like Hotel Mario? It's a bit elementary, but maybe there could be an answer simply by comparing their code.

**Comment by FoxhackDN** [comment]
> I've been trying to do a bit of extra research online, but I'm having difficulty figuring out exactly what the purpose of the DVC was even after so many years. Was it an expansion pack like the N64? Was it an add-on like the Sega CD? I'd love to try and mess around with this a bit, but the answer to those questions would definitely determine how I best approach it. I'm surprised Philips hasn't tried to help make this happen themselves. The cdi.xml makes it look like a pretty decent chunk of the library needs the DVC and since they really aren't in the console space anymore, it'd be a HUGE way to get people into it long past its EOL.

It was an optional add-on that allowed the system to play back higher quality MPEG encoded video in CDI discs and Video CDs. Like Moogly said almost a decade ago, there's no info on how it works.

Some software list items that require it WILL boot and do things if it's not installed, but they'll show an error (in the game) when they try to play back the MPEG data and keep going.

**Comment by MooglyGuy** [comment]
Honestly, if you aren't familiar with what the DVC even is or does, aren't aware that CD-i Emu is the creation of one person (CD-i Fan), and aren't aware that there has already been a significant amount of co-operation between CD-i Fan and myself in order to get MAME's CD-i driver to where it currently is, and that we don't really chat much these days because he and I have remarkably different ways of working, how can you possibly think that you're going to be able to help? Don't waste your time, but especially don't waste other peoples' with wild conjecture.

**Comment by petrij98** [comment]
> Honestly, if you aren't familiar with what the DVC even is or does, aren't aware that CD-i Emu is the creation of one person (CD-i Fan), and aren't aware that there has already been a significant amount of co-operation between CD-i Fan and myself in order to get MAME's CD-i driver to where it currently is, and that we don't really chat much these days because he and I have remarkably different ways of working, how can you possibly think that you're going to be able to help? Don't waste your time, but especially don't waste other peoples' with wild conjecture.

Well, do forgive me in wanting to help address an issue that I felt like the community would be interested in. I certainly won't be trying to trample over anyone's work done far before my time. I'm just brand new to CD-I emulation and was interested in learning more about it, especially areas where I could contribute.

**Comment by ghost** [comment]
MooglyGuy clearly did not have a good upbringing. He needs some basic lessons in politeness. He clearly does not have the intelligence to understand people who are interested and wanting to help somehow. And the worse thing for him is, politeness cannot be emulated. Such arrogance!💩

**Comment by rb6502** [comment]
petrij98 came into the thread not even knowing what the DVC is, which isn't a good basis for being helpful.  If someone wants to actually help, disassembling the loader program Meerkov pointed to would be a reasonable start.

**Comment by galibert** [comment]
It's loaded at 27fc90, you can breakpoint it at 27fce4.

  OG.

**Comment by XargonWan** [comment]
Good, the PR on MAME si merged!
